packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -214,22 +214,17 @@ export async function launch({
   verboseBlocks = false,
   metricsProvider = DEFAULT_METER_PROVIDER,
   slogSender,
-  mapSize = DEFAULT_LMDB_MAP_SIZE,
   swingStoreTraceFile,
   keepSnapshots,
   afterCommitCallback = async () => ({}),
 }) {
   console.info('Launching SwingSet kernel');
 
-  const { kvStore, streamStore, snapStore, commit } = openSwingStore(
-    kernelStateDBDir,
-    { mapSize, traceFile: swingStoreTraceFile, keepSnapshots },
-  );
-  const hostStorage = {
-    kvStore,
-    streamStore,
-    snapStore,
-  };
+  const { kernelStorage, hostStorage } = openSwingStore(kernelStateDBDir, {
+    traceFile: swingStoreTraceFile,
+    keepSnapshots,
+  });
+  const { kvStore, commit } = hostStorage;
```

Comment:
For the record, we looked into this, and `inboundQueue` is manipulated within a block, so the commit happens at the end of the block, along with everything else. It's not exactly inside a crank, but in the same way that doing a `controller.installBundle()` is not exactly inside a crank: it's inside a crank-like phase, driven by the host app, before a `controller.run()`, which does not need any sort of revert-to-start-of-crank handling.

Reply:
Yes, the inbound queue will always be manipulated during "end block", but always outside of cranks (actually outside any swingset run).

I think we can guarantee that any host access of the kvStore will always be performed outside of `controller.run()`. It might be good to add an enforcement mechanism on the host side to hold us to that guarantee.